### PR TITLE
docs: Style Guides

### PR DESCRIPTION
**Pull request by philsturgeon**
https://github.com/stoplightio/platform-docs/blob/1a6575e0edf3899888b471ff4dc73d4ae86480a9/docs/2.-governance/e.style-guides.md

### CODE REVIEW

File: docs/2.-governance/e.style-guides.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+- in [a git hook](https://stoplight.io/p/docs/gh/stoplightio/spectral/docs/guides/1-workflows.md#git-hooks)
+- in a JavaScript test suite using the [JS API](https://stoplight.io/p/docs/gh/stoplightio/spectral/docs/guides/3-javascript.md)
+- on [continuous integration](https://stoplight.io/p/docs/gh/stoplightio/spectral/docs/guides/1-workflows.md#continuous-integration) to fail builds with errors
+
+#### Distributing Rulesets
+
+Rulesets can just be defined as a file in a repository, but that will lead to every API having their own different rules, which somewhat defeats the purpose.
+
+One approach is to publish your ruleset to a publicly available URL, then folks can extend it in their repo:
```
Comment (mnaumanali94) [inline docs/2.-governance/e.style-guides.md:140]:
```suggestion
One approach is to publish your ruleset to a publicly available URL which folks can then extend in their repo:
```

File: docs/2.-governance/e.style-guides.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+except:
+  "one.yaml#":
+    - oas3-api-servers
+  "./one.yaml#/info":  # Beside relative ones, also supports rooted paths ;-)
+    - info-contact
+```
+
+
+If you already have a style guide, see how much of it you can solve with the build-in functions, then start to get more creative with custom functions to solve the organizations biggest issues.
```
Comment (mnaumanali94) [inline docs/2.-governance/e.style-guides.md:212]:
```suggestion
If you already have a style guide, see how much of it you can solve with the build-in functions, then start to get more creative with custom functions to solve the organization's biggest issues. 
```

File: docs/2.-governance/e.style-guides.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+- Ban HTTP Basic entirely
+- Make sure every endpoint has some sort of security (OAuth 2, API Key, but not both)
+- Every response should support `application/vnd.api+json` ([JSON:API](https://jsonapi.org/)) not just plain-old JSON
+- ID's as integers let people [crawl your API](https://phil.tech/http/2015/09/03/auto-incrementing-to-destruction/) incredibly easily, switch to UUID
+
+#### Errors
+
+- Your `20X` response seems to have errors in it, which is confusing
+- There are no URLs in your errors, how can anyone find out more information about what went wrong?
```
Comment (mnaumanali94) [inline docs/2.-governance/e.style-guides.md:232]:
```suggestion
- There are no URLs in your errors. How can anyone find out more information about what went wrong?
```

File: docs/2.-governance/e.style-guides.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+The trouble with these text-based documents is that they are large, terse documents which developers rarely read. If developers _do_ read them cover to cover, _and_ remember everything in there, that knowledge becomes partially out of date when new rules are added because they won't know about them until they re-read everything cover to cover again.
+
+These days style guides can be automated by programming rules into a format which robots can understand:
+
+- [Spectral](https://stoplight.io/open-source/spectral/) by [Stoplight](https://stoplight.io/)
+- [api-linter](https://github.com/googleapis/api-linter) by Google
+- [graphql-schema-linter](https://github.com/cjoudrey/graphql-schema-linter) by [Christian Joudrey](https://twitter.com/cjoudrey)
+
+We've built Spectral to be as flexible as possible for any type of JSON/YAML-based data, which includes built-in rules for OpenAPI, AsyncAPI, JSON Schema, but also means you could write custom rulesets for RAML, Kubernetes config, or anything else written in JSON or YAML.
+
```
[the remark this answers is not in the document]
Comment (philsturgeon) [reply docs/2.-governance/e.style-guides.md:55]: I took a swing at improvements there, but dunno about explorer and spectral i dont think we have any integration there.

@lottamus what do you think? also, would love a review on this in general!

File: docs/2.-governance/e.style-guides.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+          description: ok
+```
+
+Spectral will give anyone repeating an operationId a warning: 
+
+<!-- theme:warning -->
+
+> Every operation should have a unique `operationId`.
+
+Spectral avoids enforcing any opinions about the actual API being designed, but it's absolutely supported.
```
[the remark this answers is not in the document]
Comment (philsturgeon) [reply docs/2.-governance/e.style-guides.md]: Not that are good enough to link to just yet, let's make this an action item for another time.

**Pull request merged**